libs/grok/src/main/java/org/elasticsearch/grok/Grok.java

```diff
@@ -163,48 +164,53 @@ public String groupMatch(String name, Region region, String pattern) {
      * @return named regex expression
      */
     public String toRegex(String grokPattern) {
-        byte[] grokPatternBytes = grokPattern.getBytes(StandardCharsets.UTF_8);
-        Matcher matcher = GROK_PATTERN_REGEX.matcher(grokPatternBytes);
+        StringBuilder res = new StringBuilder();
+        int i = 0;
+        while (true) {
```

Comment:
consider changing this to `for` loop instead of `while` and moving exception to the end of the method
```java
for (int i = 0; i < MAX_TO_REGEX_ITERATIONS; i++) {
    ...
}
throw new IllegalArgumentException("Can not convert grok patterns to regular expression");
```